File: src/exam-core/src/main/java/com/wcpdoc/exam/core/entity/Question.java
package com.wcpdoc.exam.core.entity;

import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.wcpdoc.core.mybatis.IntTypeHandler;

import lombok.Data;

/**
 * 试题实体
 * 
 * v1.0 zhanghc 2017-05-07 14:56:29
 */
@Data
@TableName(value = "EXM_QUESTION", autoResultMap = true)
public class Question {
	@TableId(type = IdType.AUTO)
	private Integer id;
	private Integer type;
	private String title;
	private Integer markType;
	@TableField(typeHandler = IntTypeHandler.class)
	private List<Integer> markOptions;
	private String analysis;
	private BigDecimal score;
	private Integer state;
	private Integer questionTypeId;
	private Integer createUserId;
	private Integer updateUserId;
	private Date updateTime;
}

File: h5/src/ts/exam/question.ts
export interface Question {
    id?: number | null // 试题ID
    type: number// 试题类型 （1：单选；2：多选；3：填空；4：判断；5：问答）
    title: string// 标题
    options: string[]// 选项
    markType: number// （阅卷类型 1：客观题；2：主观题）
    markOptions: number[] //（2：答案无顺序；3：不区分大小写；)
    score: number// 分数
    answers: string[]// 答案
    scores: number[]// 答案分数（单选判断主观问答，只有第0项有效）
    analysis: string// 解析
    questionBankId?: number | null // 题库ID
    [key: string]: any; // 扩展字段
}
